### PR TITLE
Many JS updates

### LINKED ISSUE #51: Javascript updates
https://github.com/mitodl/micromasters/pull/252
https://github.com/mitodl/micromasters/pull/247
https://github.com/mitodl/micromasters/pull/340
https://github.com/mitodl/micromasters/pull/354
https://github.com/mitodl/micromasters/pull/263
https://github.com/mitodl/micromasters/pull/264
https://github.com/mitodl/micromasters/pull/246
https://github.com/mitodl/micromasters/pull/156/

Also update package requirements according to the latest on micromasters
Also update eslint rules

### PR DESCRIPTION
**Pull request by noisecapella**
Fixes #51

### CODE REVIEW

**Comment by pdpinch** [comment]
@noisecapella is there an issue that this goes with?

**Comment by aliceriot** [comment]
@noisecapella should I just read the code, or is there some manual testing I should do? I see that the CI build passed.

File: {{ cookiecutter.project_name }}/package.json [hunk 1]
```diff
@@ -2,45 +2,54 @@
...
     "redux-logger": "2.6.1",
     "redux-thunk": "2.0.1",
-    "sass-loader": "3.1.2",
-    "style-loader": "0.13.0",
-    "webpack": "1.12.14",
+    "sass-loader": "3.2.0",
+    "sinon": "1.17.4",
+    "style-loader": "0.13.1",
+    "url-loader": "0.5.7",
+    "webpack": "1.13.0",
```
Comment (aliceriot) [inline {{ cookiecutter.project_name }}/package.json]: would it make sense to add Flow in this PR, or should we hold off for now?

**Comment by noisecapella** [comment]
@pdpinch I'll create one and edit the description

@aliceriot The manual testing is to just click the checkbox, there's not much else right now. It's mostly just code review, it should resemble the changes we made to micromasters

**Comment by aliceriot** [comment]
this all looks good to me, the only question is if it makes sense to update this PR with JS changes we've made in the meantime (flow, various supporting libraries for flow, etc).

File: {{ cookiecutter.project_name }}/package.json (hunk 1 above)
Comment (noisecapella) [inline {{ cookiecutter.project_name }}/package.json]: I think a separate PR would be cleaner, and we should get flow running as part of the micromasters CI first, then do that as part of the work here too.

Comment (aliceriot) [inline {{ cookiecutter.project_name }}/package.json]: sounds good to me!

**Comment by aliceriot** [comment]
:+1:

**Pull request merged**